src/main/java/dev/meyi/noscamspam/checks/UserCheck.java

```diff
@@ -13,40 +18,51 @@
 
 public class UserCheck {
 
-  private static final CloseableHttpClient httpClient = HttpClientBuilder.create().build();
-
-  public static boolean shouldBlockUser(String username) {
-    if (NoScamSpam.config.whitelist.contains(username)) {
-      return false;
-    } else if (NoScamSpam.blacklist.contains(username)) {
-      return true;
-    }
+    private static final CloseableHttpClient httpClient = HttpClientBuilder.create().build();
 
-    try (CloseableHttpResponse response = httpClient.execute(new HttpGet(
-        "https://api.hypixel.net/player?key=" + NoScamSpam.config.apiKey + "&name=" + username));
-        InputStream stream = response.getEntity().getContent()) {
-      StatsResponse statsResponse = NoScamSpam.gson
-          .fromJson(IOUtils
-              .toString(stream,
-                  StandardCharsets.UTF_8), StatsResponse.class);
-
-      if (statsResponse.success) {
-        if (statsResponse.player.networkExp > NoScamSpam.config.hypixelExperience &&
-            statsResponse.player.achievements.getSkillAverage() > NoScamSpam.config.skillAverage &&
-            statsResponse.player.achievements.skyblock_dungeoneer
-                > NoScamSpam.config.catacombsLevel) {
-          return false;
-        } else {
-          if (!NoScamSpam.blacklist.contains(username)) {
-            NoScamSpam.blacklist.add(username);
-          }
-          return true;
+    public static void shouldBlockUser(String username, ClientChatReceivedEvent event) {
```

Comment:
Is there any specific reason to make this `void` and do the event canceling here rather than in the event handler itself?

Reply:
Also, are forge events even blocking in the first place? It may just be my connection so I never notice it, but I haven't seen any game freezes

Reply:
No there is not, I will move this straight to the event